woocommerce/woocommerce
# refactor: `*_checkout_process_payment_error` to return 400 error code

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





`402` error codes shouldn't be returned by any WP API.
More info here: p9F6qB-cgE-p2#comment-58934
Discussed internally here: p1711358609738689-slack-C02TS23QJ1X

Changing the `402` error code to `400`. This is basically a revert of https://github.com/woocommerce/woocommerce-blocks/pull/7273.
Unfortunately, there isn't a graceful way to deprecate the error code.

Closes https://github.com/Automattic/woocommerce-payments/issues/8080



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

In order to test this, I created a "dummy" gateway that always fails when selected to process a payment:
[woocommerce-failed-gateway.zip](https://github.com/woocommerce/woocommerce/files/14770589/woocommerce-failed-gateway.zip)


1. Create and configure a WC site
2. Activate and enable the gateway plugin linked above
3. Ensure your "checkout" page uses WC checkout block
4. Add products to the cart, go to checkout
5. Select the "Failed gateway" gateway that you enabled earlier, click "Place order"
6. Look at your network tab in the browser's dev tools
7. The `wc/store/v1/checkout ` endpoint should fail with a `400` error code (instead of `402`)

Before:
<img width="1406" alt="Screenshot 2024-03-27 at 9 53 30 AM" src="https://github.com/woocommerce/woocommerce/assets/273592/e81506be-4b4d-4c3b-a69c-a332dc3fd731">

After:
<img width="1363" alt="Screenshot 2024-03-27 at 9 55 16 AM" src="https://github.com/woocommerce/woocommerce/assets/273592/f981bff9-b1b3-462c-93aa-e6ae0aed6262">




### Changelog entry




<details>

#### Significance




#### Type




#### Message 
refactor: `woocommerce_rest_checkout_process_payment_error` returns a `400` response code, instead of `402`.
#### Comment 

</details>
